test/prototype_common_utils.py

```diff
@@ -69,6 +70,7 @@ def compare(self) -> None:
         self._compare_attributes(actual, expected)
 
         actual, expected = self._equalize_attributes(actual, expected)
+        actual, expected = self._promote_for_comparison(actual, expected)
```

Comment:
This was a bug. Without this promotion, we are later subtracting `uint8` values for the absolute difference. They wrap around:

```pycon
>>> a = torch.tensor(10, dtype=torch.uint8)
>>> b = torch.tensor(11, dtype=torch.uint8)
>>> a - b
tensor(255, dtype=torch.uint8)
>>> abs(a - b)
tensor(255, dtype=torch.uint8)
```

If we convert to a singed integer dtype first (here `int64`) we get the correct absolute diff:

```pycon
>>> a = torch.tensor(10, dtype=torch.int64)
>>> b = torch.tensor(11, dtype=torch.int64)
>>> a - b
tensor(-1, dtype=torch.int64)
>>> abs(a - b)
tensor(1, dtype=torch.int64)
```

This bug only affected the comparisons that used `agg_method=...`.